src/resolvers.ts

```diff
@@ -0,0 +1,130 @@
+import { Book} from "./models/book";
+import { Customer } from "./models/customer";
+import { LendingRecord } from "./models/lendingRecord";
+import { BooksDataSource } from "./dataSources/booksDataSource";
+import { CustomersDataSource } from "./dataSources/customersDataSource";
+import { logger } from "./logger";
+
+const booksDataSource = new BooksDataSource();
+const customersDataSource = new CustomersDataSource();
+
+export const resolvers = {
```

Comment:
I made the code asynchronous. About the deadlock I don't think that a deadlock may occur since I am not using a lock and I am using now asynchronous read/write files operations. Furthermore If  I remember correctly we don't have many threads here and we can't have a situation where one thread locks a resource and depends on another one.